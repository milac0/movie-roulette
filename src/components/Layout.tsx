import React from "react";
import { css } from "emotion";
import Header from "./Header";
import Footer from "./Footer";

const container = css`
  display: flex;
  flex-direction: column;
  min-height: 100vh;
`;

const content = css`
  flex-grow: 1;
`;

interface Props {
  children: React.ReactNode;
}

const Layout: React.FC<Props> = ({ children }) => {
  return (
    <div className={container}>
      <div className={content}>
        <Header />
        {children}
      </div>
      <Footer />
    </div>
  );
};

export default Layout;
