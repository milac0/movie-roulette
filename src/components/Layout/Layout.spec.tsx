import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Layout from './Layout';

describe('Hover renders correctly', () => {
    test('snapshot renders', () => {
        const child = <div>testing div!</div>

        const tree = renderer.create(
            <Router>
                <Layout>
                    {child}
                </Layout>
            </Router>
        ).toJSON();
        expect(tree).toMatchSnapshot();
    });
});

