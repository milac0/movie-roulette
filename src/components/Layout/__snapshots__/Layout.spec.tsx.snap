// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hover renders correctly snapshot renders 1`] = `
<div
  className="container"
>
  <div
    className="content"
  >
    <div
      className="header"
    >
      <div
        className="headerContent"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <h1>
            Movie roulette
          </h1>
        </a>
        <button
          className="btn"
          onClick={[Function]}
        >
          Login as Guest
        </button>
      </div>
    </div>
    <div>
      testing div!
    </div>
  </div>
  <div
    className="footer"
  >
    <p>
      API provided by
       
      <a
        href="https://www.themoviedb.org/"
        target="_blank"
      >
        TMDb
      </a>
    </p>
  </div>
</div>
`;
